docs/Formula-Android.md

```diff
@@ -0,0 +1,380 @@
+```kotlin
+/**
+ * Fragment key has to provide Parcelable implementation because it is passed 
+ * to the fragment as an argument. 
+ *   
+ * Read more about Parcelize: https://kotlinlang.org/docs/tutorials/android-plugin.html
+ */
+@Parcelize
+data class CounterKey(
+    override val tag: String = "counter"
+) : FragmentKey
+```
+
+### Define a feature factory
+Feature factory creates state observable and a view factory for a fragment with a 
+specific `FragmentKey` type. To continue our example, we define a `CounterFeatureFactory` 
+which will handle `CounterKey` fragments.
+```kotlin
+class CounterFeatureFactory : FeatureFactory<Any, CounterKey> {
+    override fun initialize(dependencies: Any, key: CounterKey): Feature<*> {
+        val counterFormula = CounterFormula()        
+        return Feature(
+            state = counterFormula.toObservable(),
+            viewFactory = CounterViewFactory()
+        )   
+    }
+}
+
+// View factory which uses XML layout resource.
+class CounterViewFactory : LayoutViewFactory<CounterRenderModel>(R.layout.counter) {
+    override fun ViewInstance.create(): FeatureView<CounterRenderModel> {
+        // We use [ViewInstance.view] to access the inflated view
+        val counterView = CounterRenderView(view)
+
+        // We create a [FeatureView] by passing a [RenderView]
+        return featureView(counterView)
+    }
+}
+```
+
+We now need to register our feature factory with the activity in which the counter will be shown in.
+```kotlin
+class MyApp : Application() {
+    
+    override fun onCreate() {
+        super.onCreate()
+        
+        FormulaAndroid.init(this) {
+            activity(MyActivity::class) {
+                store {
+                    bind(CounterFeatureFactory())
+                }
+            }
+        }
+    }
+}
+```
+
+### Use formula fragment
+The only thing left is navigating to this screen. We create `FormulaFragment` instance using
+our `CounterKey` and use fragment transactions to add it.
+```kotlin
+class MyActivity : FormulaAppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.my_activity)
+        
+        if (savedInstanceState == null) {
+            val key = CounterKey()
+            val fragment = FormulaFragment.newInstance(key)
+            
+            // Add the fragment using the fragment transaction API.
+            supportFragmentManager.beginTransaction()
+                .add(R.id.activity_content, fragment, key.tag)            
+                .commit()    
+        }
+    }
+}
+```
+
+If your `Activity` has another base class, you can just copy logic from `FormulaAppCompatActivity` into your `Activity`.
+
+### And that's it
+Formula takes care of the rest. The RxJava state stream is instantiated and subscribed to when the user enters 
+declared navigation destination. We dispose of the stream only when user exits the destination. 
```

Comment:
Maybe a mention of what happens if the screen were to be rotated or another configuration change happens.

Reply:
Updated to 
> Formula takes care of the rest. This is how the state observable works:
> - When FormulaFragment is added we instantiate and subscribe to the state observable.
> - When FormulaFragment is removed we destroy the state observable.
> 
> The state management observable continues to run during configuration changes or if you navigate forward.